src/lib.rs

```diff
@@ -4345,11 +4388,23 @@ impl Connection {
 
             frame::Frame::PathResponse { .. } => (),
 
-            frame::Frame::ConnectionClose { .. } => {
+            frame::Frame::ConnectionClose {
+                error_code, reason, ..
```

Comment:
If you destructure this as `reason_phrase` then it simplifies the creation of `ConnectionError` below. Some for the ApplicationClose case below.